wrappers/services_test.go

```diff
@@ -189,6 +189,120 @@ apps:
 	})
 }
 
+func (s *servicesTestSuite) TestServicesEnableState(c *C) {
+	info := snaptest.MockSnap(c, packageHello+`
+ svc2:
+  command: bin/hello
+  daemon: forking
+`, &snap.SideInfo{Revision: snap.R(12)})
+	svc1File := "snap.hello-snap.svc1.service"
+	svc2File := "snap.hello-snap.svc2.service"
+
+	s.systemctlRestorer()
+	r := testutil.MockCommand(c, "systemctl", `#!/bin/sh
+	if [ "$1" = "--root" ]; then
+		# shifting by 2 also drops the temp dir arg to --root
+	    shift 2
+	fi
+
+	case "$1" in
+		is-enabled)
+			case "$2" in 
+			"snap.hello-snap.svc1.service")
+				echo "disabled"
+				exit 1
+				;;
+			"snap.hello-snap.svc2.service")
+				echo "enabled"
+				exit 0
+				;;
+			*)
+				echo "unexpected service $*"
+				exit 2
+				;;
+			esac
+	        ;;
+	    *)
+	        echo "unexpected op $*"
+	        exit 2
+	esac
+
+	exit 1
+	`)
+	defer r.Restore()
+
+	states, err := wrappers.ServicesEnableState(info, progress.Null)
+	c.Assert(err, IsNil)
+
+	c.Assert(states, DeepEquals, map[string]bool{
+		"svc1": false,
+		"svc2": true,
+	})
+
+	// the calls could be out of order in the list, since iterating over a map
+	// is non-deterministic, so manually check each call
+	c.Assert(r.Calls(), HasLen, 2)
+	for _, call := range r.Calls() {
+		c.Assert(call, HasLen, 5)
+		c.Assert(call[:4], DeepEquals, []string{"systemctl", "--root", s.tempdir, "is-enabled"})
+		switch call[4] {
+		case svc1File, svc2File:
+		default:
+			c.Errorf("unknown service for systemctl call: %s", call[4])
+		}
+	}
+}
+
+func (s *servicesTestSuite) TestServicesEnableStateFail(c *C) {
+	info := snaptest.MockSnap(c, packageHello, &snap.SideInfo{Revision: snap.R(12)})
+	svc1File := "snap.hello-snap.svc1.service"
+
+	s.systemctlRestorer()
+	r := testutil.MockCommand(c, "systemctl", `#!/bin/sh
+	if [ "$1" = "--root" ]; then
```

Comment:
Given that this is only called once and we always want to return an error we can maybe simplify this to something like:
```
#!/bin/sh
echo "whoops"
exit 1
```
wdyt?

Reply:
well I still think it's useful to catch programming errors, for example if for some reason we don't call `systemctl` with `--root`, this test would fail with "unexpected op ..." and it would be clear that there's something more unexpected going on here, so I think I'll keep it as is